modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -19,6 +19,11 @@ import (
 	host "github.com/cosmos/ibc-go/v5/modules/core/24-host"
 )
 
+// MessageRouter ADR 031 request type routing
+type MessageRouter interface {
```

Comment:
Instead of having the interface defined twice, maybe we could add it to `modules/apps/27-interchain-accounts/types`? I guess t doesn't really belong to `expected_keepers.go`, so maybe we need a new file for it...?

Reply:
Agreed, I extracted it out to router.go. Let me know if you need anything else :).